src/server/cluster/cluster_slot_migration.h

```diff
@@ -45,6 +46,9 @@ class ClusterSlotMigration : ProtocolClient {
   // Creates flows, one per shard on the source node and manage migration process
   std::error_code InitiateSlotsMigration();
 
+  // should be called after we finish all flows
+  bool IsFinalized() const;
```

Comment:
I'm not a fan of this method name, as it sounds 'constant' but in fact has a side effect (calling `Stop()`).
How about renaming it to `StopIfFinalized()`, which has the same return value?
Or keep it not mutating, and add another `Stop()` method?